src/plugins/paste-image/index.ts

```diff
@@ -0,0 +1,42 @@
+import { PluginController } from "../PluginController";
+
+export default class PasteImage extends PluginController {
+  static id = "paste-image" as const;
+  static enabledByDefault = true;
+
+  pasteHandler = this._pasteHandler.bind(this);
+
+  afterEnable() {
+    document.addEventListener("paste", this.pasteHandler);
+  }
+
+  afterDisable() {
+    document.removeEventListener("paste", this.pasteHandler);
+  }
+
+  _pasteHandler(e: ClipboardEvent) {
+    if (!this.calc.selectedExpressionId) {
+      // avoid images being inserted at the top of the expressions list
+      // when there is no selected expression
+      if (document.activeElement !== document.body) return;
+      const lastExpression: Desmos.ExpressionState = this.calc
+        .getExpressions()
+        .splice(-1)[0];
+      this.cc.dispatch({
```

Comment:
replacing this dispatch event with `dispatch({ type: "new-expression-at-end" })` might be better?
removes the need for `lastExpression` and prevents a selection bug

need to add it as a valid event in `./Calc.ts` though

Reply:
That way is simpler, but that sometimes makes strange the way of inserting images.
- It inserts new expression even when the last expression is empty
- When inserting images, the insertion location is determined for each, rather than following the first insertion location inside `_toplevelNewItemAtSelection`.  In the method of using `"new-expression-at-end"`, the second and subsequent images are inserted at the top due to the specification that inserting an image with an empty expression selected replaces the expression with the image and then deselect.

![example](https://github.com/user-attachments/assets/4c94b6d0-35d9-419e-a530-f4d82fb74443)

I noticed during this process the bug that images aren't inserted when `showIDs` is enabled and `getExpression()` returns `[]`.

Reply:
Sorry I guess I was mistaken. The latter also happens when using `lastExpression`... Anyway, I found a solution to the problem of the insertion of multiple images, but there are still a few problems. I'd like some more time.